### PR TITLE
feat: image loading state

### PR DESCRIPTION
**Pull request by csiyang**
# Description
Adding loading circles when updating an image and waiting for it to load.

[- Link to Basecamp Todo](https://3.basecamp.com/3105655/buckets/26787371/todos/4752779793)

# How should this PR be QA Tested?
- [ ] Creating and updating an image should show a loading circle on the dialog that updates it.

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/VideoBlockEditorSettingsPoster.spec.tsx [hunk 1]
```diff
@@ -52,6 +95,8 @@ const image: ImageBlock = {
   blurhash: ''
 }
 
+// const onClose = jest.fn()
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/VideoBlockEditorSettingsPoster.spec.tsx:98]: Remove comment

File: apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/VideoBlockEditorSettingsPoster.tsx [hunk 2]
```diff
@@ -49,7 +253,10 @@ export function VideoBlockEditorSettingsPoster({
         borderRadius={2}
       >
         <Stack direction="row" justifyContent="space-around">
-          <ImageBlockThumbnail selectedBlock={selectedBlock} />
+          <ImageBlockThumbnail
+            selectedBlock={selectedBlock}
+            loading={loading}
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/VideoBlockEditorSettingsPoster.tsx:258]: Moved from everything from poster dialog into here to allow this imageBlockThumbnail to also have the loading state

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Card/BackgroundMedia/Image/BackgroundMediaImage.tsx [hunk 3]
```diff
@@ -90,18 +90,25 @@ export function BackgroundMediaImage({
...
 
+  const [loading, setLoading] = useState(false)
+  useEffect(() => {
+    if (createLoading || updateLoading) {
+      setLoading(true)
+    } else {
+      setLoading(false)
+    }
+  }, [createLoading, updateLoading])
+
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Card/BackgroundMedia/Image/BackgroundMediaImage.tsx:111]: just use createLoading || updateLoading

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Card/BackgroundMedia/Image/BackgroundMediaImage.tsx [hunk 4]
```diff
@@ -250,6 +257,7 @@ export function BackgroundMediaImage({
       selectedBlock={imageBlock}
       onChange={handleChange}
       onDelete={handleImageDelete}
+      loading={loading}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Card/BackgroundMedia/Image/BackgroundMediaImage.tsx:260]:
```suggestion
      loading={createLoading || updateLoading}
```

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Image/Options/ImageOptions.tsx [hunk 5]
```diff
@@ -32,9 +32,19 @@ export function ImageOptions(): ReactElement {
...
   const imageBlock = selectedBlock as TreeBlock<ImageBlock>
 
+  const [loading, setLoading] = useState(false)
+  useEffect(() => {
+    if (updateLoading) {
+      setLoading(true)
+    } else {
+      setLoading(false)
+    }
+  }, [updateLoading])
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Image/Options/ImageOptions.tsx:46]: remove.

File: apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Image/Options/ImageOptions.tsx [hunk 6]
```diff
@@ -32,9 +32,19 @@ export function ImageOptions(): ReactElement {
   } = useEditor()
   const journey = useJourney()
   const { enqueueSnackbar } = useSnackbar()
-  const [imageBlockUpdate] = useMutation<ImageBlockUpdate>(IMAGE_BLOCK_UPDATE)
+  const [imageBlockUpdate, { loading: updateLoading }] =
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Image/Options/ImageOptions.tsx:35]:
```suggestion
  const [imageBlockUpdate, { loading }] =
```

File: apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/ImageEdit/ImageEdit.tsx [hunk 7]
```diff
@@ -66,19 +66,26 @@ export function ImageEdit(): ReactElement {
...
 
+  const [loading, setLoading] = useState(false)
+  useEffect(() => {
+    if (createLoading || updateLoading) {
+      setLoading(true)
+    } else {
+      setLoading(false)
+    }
+  }, [createLoading, updateLoading])
+
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/ImageEdit/ImageEdit.tsx:88]: remove

File: apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/ImageEdit/ImageEdit.tsx [hunk 8]
```diff
@@ -277,6 +284,7 @@ export function ImageEdit(): ReactElement {
           selectedBlock={journey?.primaryImageBlock ?? null}
           onChange={handleChange}
           onDelete={handleDelete}
+          loading={loading}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/Drawer/SocialShareAppearance/ImageEdit/ImageEdit.tsx:287]:
```suggestion
          loading={createLoading || updateLoading}
```

**Review by tataihono — CHANGES_REQUESTED** [review]
remove an intermediary state and just use the original state. you are causing rerenders when its not necessary.

File: apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/Dialog/VideoBlockEditorSettingsPosterDialog.tsx [hunk 9]
```diff
@@ -83,14 +83,28 @@ export function VideoBlockEditorSettingsPosterDialog({
...
+  ] = useMutation<PosterImageBlockUpdate>(POSTER_IMAGE_BLOCK_UPDATE)
+
+  const [loading, setLoading] = useState(false)
+  useEffect(() => {
+    if (createLoading || updateLoading) {
+      setLoading(true)
+    } else {
+      setLoading(false)
+    }
+  }, [createLoading, updateLoading])
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/Dialog/VideoBlockEditorSettingsPosterDialog.tsx:107]: delete and use `createLoading || updateLoading` below

File: apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/Dialog/VideoBlockEditorSettingsPosterDialog.tsx [hunk 10]
```diff
@@ -220,6 +234,7 @@ export function VideoBlockEditorSettingsPosterDialog({
         selectedBlock={selectedBlock}
         onChange={handleChange}
         onDelete={deleteCoverBlock}
+        loading={loading}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/Dialog/VideoBlockEditorSettingsPosterDialog.tsx:237]:
```suggestion
        loading={createLoading || updateLoading}
```

File: apps/journeys-admin/src/components/Editor/ImageBlockEditor/ImageBlockEditor.spec.tsx [hunk 11]
```diff
@@ -26,6 +26,7 @@ describe('ImageBlockEditor', () => {
           selectedBlock={null}
           onChange={onChange}
           onDelete={onDelete}
+          loading={false}
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ImageBlockEditor/ImageBlockEditor.spec.tsx:29]: the underlying component should not make this boolean required negating the need to add this everywhere.

File: apps/journeys-admin/src/components/Editor/ImageBlockEditor/ImageBlockEditor.tsx [hunk 12]
```diff
@@ -15,13 +15,15 @@ interface ImageBlockEditorProps {
   showDelete?: boolean
   onChange: (block: ImageBlock) => Promise<void>
   onDelete?: () => Promise<void>
+  loading: boolean
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ImageBlockEditor/ImageBlockEditor.tsx:18]:
```suggestion
  loading?: boolean
```

File: apps/journeys-admin/src/components/Editor/ImageBlockHeader/ImageBlockHeader.tsx [hunk 13]
```diff
@@ -12,14 +12,16 @@ interface ImageBlockHeaderProps {
   header: string
   showDelete: boolean
   onDelete: () => Promise<void>
+  loading: boolean
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ImageBlockHeader/ImageBlockHeader.tsx:15]:
```suggestion
  loading?: boolean
```

File: apps/journeys-admin/src/components/Editor/ImageBlockThumbnail/ImageBlockThumbnail.tsx [hunk 14]
```diff
@@ -24,7 +27,9 @@ export function ImageBlockThumbnail({
         overflow: 'hidden'
       }}
     >
-      {selectedBlock?.src != null ? (
+      {loading ? (
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/ImageBlockThumbnail/ImageBlockThumbnail.tsx:30]:
```suggestion
      {loading === true ? (
```

File: apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/VideoBlockEditorSettingsPoster.spec.tsx [hunk 15]
```diff
@@ -39,7 +39,6 @@ const video: VideoBlock = {
   },
   posterBlockId: null
 }
-
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/VideoBlockEditorSettingsPoster.spec.tsx:42]: undo

File: apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/VideoBlockEditorSettingsPoster.tsx [hunk 16]
```diff
@@ -15,7 +15,6 @@ interface BackgroundMediaCoverImageProps {
   parentBlockId: string | undefined
   disabled?: boolean
 }
-
```
Comment (tataihono) [inline apps/journeys-admin/src/components/Editor/VideoBlockEditor/Settings/Poster/VideoBlockEditorSettingsPoster.tsx:18]: undo

**Review by tataihono — CHANGES_REQUESTED** [review]
you can remove a number of the loading={false} statements if you just make the loading boolean optional. Because there is no behavioral adjustments to the component when it is false all you really care about is if loading is true.

**Review by tataihono — APPROVED** [review]

**Pull request merged**